eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -339,7 +339,7 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
           val localFundingPubkey = keyManager.fundingPublicKey(localParams.channelKeyPath).publicKey
           val fundingPubkeyScript = Script.write(Script.pay2wsh(Scripts.multiSig2of2(localFundingPubkey, remoteParams.fundingPubKey)))
           wallet.makeFundingTx(fundingPubkeyScript, Satoshi(fundingSatoshis), fundingTxFeeratePerKw).pipeTo(self)
-          goto(WAIT_FOR_FUNDING_INTERNAL) using DATA_WAIT_FOR_FUNDING_INTERNAL(temporaryChannelId, localParams, remoteParams, fundingSatoshis, pushMsat, initialFeeratePerKw, accept.firstPerCommitmentPoint, open)
+          goto(WAIT_FOR_FUNDING_INTERNAL) using DATA_WAIT_FOR_FUNDING_INTERNAL(temporaryChannelId, localParams, remoteParams, Satoshi(fundingSatoshis), pushMsat, initialFeeratePerKw, accept.firstPerCommitmentPoint, open)
```

Comment:
Same here, fundingSatoshis is supposed to be typed ?

Reply:
It will be typed but that change shouldn't be here **now**. To clarify i applied the following pattern when typing the amounts in `wire` messages:
- When doing a comparison between typed a non typed amounts use raw data types ( Long vs Long)
- Changes should applied only to `wire` messages, other classes using the same fields (i.e `LocalParams` have remained unchanged.
- Classes that use the new typed amounts have been migrated to typed only if all the fields are now typed , [example](https://github.com/ACINQ/eclair/pull/1088/files#diff-b90a51e9dd7869ff814b488a4271daad)
- Functions that mimic constructor also pulled the change from the underlying type [example](https://github.com/ACINQ/eclair/pull/1088/files#diff-d57afc529bb6dd213ae17d552113061dR40)

So the change you commented shouldn't be there, `DATA_WAIT_FOR_FUNDING_INTERNAL.fundingSatoshis` should stay `Long` for the moment. Alternatively i can propagate the typing (and diverge from the PR-wide heuristics) to this class and type "pushMsat" as well.